### PR TITLE
add alerts history with filters TKT-3002

### PR DESCRIPTION
**Pull request by ofirit**

### CODE REVIEW

File: intezer_sdk/alerts.py [hunk 1]
```diff
@@ -40,6 +46,221 @@ def get_alerts_by_alert_ids(alert_ids: List[str],
...
+        filters['exclude_alert_ids'] = exclude_alert_ids
+    if usernames:
+        filters['usernames'] = usernames
+    if file_hashes:
+        filters['file_hashes'] = file_hashes
+    if process_commandlines:
+        filters['process_commandlines'] = process_commandlines
+    if sort_by:
+        filters['sort_by'] = sort_by
+    if is_mitigated:
```
Comment (almogch) [inline intezer_sdk/alerts.py:121]: think of adding a help func to add key in not None

```
def _add_filter(filters, key, value):
  if value:
    filter[key] = value
```

File: intezer_sdk/alerts.py [hunk 2]
```diff
@@ -40,6 +46,221 @@ def get_alerts_by_alert_ids(alert_ids: List[str],
...
+    :param email_subject: Query alerts only with this email subject.
+    :param email_cc: Sort alerts only with this email cc.
+    :param email_bcc: Sort alerts only with this email bcc.
+    :param email_message_id: Sort alerts only with this email message id.
+    :param email_reported_by: Sort alerts only with this email reported by.
+    :param device_private_ips: Query alerts only with these private ips.
+    :param device_external_ips: Query alerts only with these external ips.
+    :param device_ids: Query alerts only with these device ids.
+    :param time_filter_type: The time value to filter alerts by.
+    :param sort_order: The order to sort the alerts by.
```
Comment (almogch) [inline intezer_sdk/alerts.py:220]: explain what can be the values here

File: intezer_sdk/alerts.py [hunk 3]
```diff
@@ -40,6 +46,221 @@ def get_alerts_by_alert_ids(alert_ids: List[str],
...
+    :param email_recipient: Query alerts only with these email recipient.
+    :param email_subject: Query alerts only with this email subject.
+    :param email_cc: Sort alerts only with this email cc.
+    :param email_bcc: Sort alerts only with this email bcc.
+    :param email_message_id: Sort alerts only with this email message id.
+    :param email_reported_by: Sort alerts only with this email reported by.
+    :param device_private_ips: Query alerts only with these private ips.
+    :param device_external_ips: Query alerts only with these external ips.
+    :param device_ids: Query alerts only with these device ids.
+    :param time_filter_type: The time value to filter alerts by.
```
Comment (almogch) [inline intezer_sdk/alerts.py:219]: what does it mean?

File: intezer_sdk/alerts.py [hunk 4]
```diff
@@ -40,6 +46,221 @@ def get_alerts_by_alert_ids(alert_ids: List[str],
...
+    :param process_commandlines: Query alerts only with these process commandlines.
+    :param sort_by: Sort alerts only with this sort_by_key value.
+    :param is_mitigated: Query alerts only with this is_mitigated value.
+    :param email_sender: Query alerts only with these email sender.
+    :param email_recipient: Query alerts only with these email recipient.
+    :param email_subject: Query alerts only with this email subject.
+    :param email_cc: Sort alerts only with this email cc.
+    :param email_bcc: Sort alerts only with this email bcc.
+    :param email_message_id: Sort alerts only with this email message id.
+    :param email_reported_by: Sort alerts only with this email reported by.
```
Comment (almogch) [inline intezer_sdk/alerts.py:215]: sort?

**Review by almogch — APPROVED** [review]

File: intezer_sdk/alerts_results.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+class AlertsHistoryResult(HistoryResult):
+
+    def __init__(self, request_url_path: str, api: IntezerApiClient, filters: Dict):
+        """
+        Fetch all alerts history results from server.
+        """
+        super().__init__(request_url_path, api, filters)
+
+    def _fetch_history(self, url_path: str, data: Dict
+                       ) -> Tuple[int, List]:
```
Comment (yoniabrahamy) [inline intezer_sdk/alerts_results.py:19]: Is this how the line should break? Do we use Black here?

**Review by yoniabrahamy — APPROVED** [review]

File: intezer_sdk/alerts_results.py (hunk 5 above)
Comment (ofirit) [reply intezer_sdk/alerts_results.py:19]: No :\ will fix it (again)

**Pull request merged**